docs: add 'Next Steps' in 'Getting Started'

HaramChoi-askui opened the pull request:
1. Added a new page 'Next Steps' in 'Getting Started' that suggests reading materials after the setup.

2. Re-arranged the sub-items in the sidebar:

    - Tooling/annotate-image.md -> Getting_Started/annotate-image.md

    - Tooling/annotate-interactively.md -> Getting_started/annotate-interactively.md

    - Best_Practice/configure_test_framework.md -> Tooling/configure_test_framework.md

JohannesDienst-askui commented on docs/docs/general/02-Getting Started/annotate-interactively.md:
```diff
@@ -1,10 +1,17 @@
+
+📌 *Press `esc` to exit the full-screen window.*
 
 ![Interactive Annotation](./interactive-annotate.gif)
```
Please move the gif to `static/img/gif` to save repository space.

Otherwise, they would be copied with every version 😉

JohannesDienst-askui commented:
I wonder if we could fuse the pages `annotate-image` and `annotate-interactively` and rename it to something more helpful. As a user, I would not know why I should click that when I search for help on how to select/interact with a specific element.

Maybe something like: _Debug Element Selection_

HaramChoi-askui commented:
I have seen that @JohannesDienst-askui appended a compactly nice chapter `where to go next?` to `write your first automation`, but still, I would like to suggest to have a discrete page for the `next steps`.

HaramChoi-askui commented:
I have seen that @JohannesDienst-askui appended a compactly nice chapter `where to go next?` to `write your first automation`, but still, I would like to suggest to have a discrete page for the `next steps`.

HaramChoi-askui commented:
I have seen that @JohannesDienst-askui appended a compact and nice chapter `where to go next?` to `write your first automation`, but still, I would like to suggest to have a discrete page for the `next steps`.

I also would like to suggest to re-categorise and re-arrange the articles in our docs as the following change:

1. Added `next-steps.md` in `02-Getting Started`
2. Added `selecting-by-visual-relation.md and `selecting-with-text.md` in `03-Best Practice`
3. Removed `Selecting UI Elements` from `03-Best Practice`
5. Moved `Configure Test Frameworks` to `05-Tooling`
6. Merged `Annotate Image` and `Annotate Interactively` to `debug-with-annotation.md`
7. Added additional gifs for `selecting-with-text.md`

HaramChoi-askui commented:
@JohannesDienst-askui One more thing that I would love to consider regarding the re-arrangement.

HaramChoi-askui commented:
@JohannesDienst-askui
There is one more thing that I would love to discuss regarding the re-arrangement.
- **Creating sub-sections for the API. For an instance:**

Commands
    ㅏ Keyboard
    ㅏ Mouse
     ㄴ etc.

Filters
    ㅏ Text
    ㅏ Interactive (button, checkbox, radio button, etc.)
    ㅏ Table ? (row, column, etc)
     ㄴ etc.


But we shall first discuss about reducing the API list.

JohannesDienst-askui commented on docs/docs/general/03-Best Practice/selecting-with-text.md:
```diff
@@ -0,0 +1,173 @@
+# Selecting an Element with Text
```
This is an awesome page. Do we have a blogpost already? It could definitely be converted to one!

JohannesDienst-askui commented on docs/docs/general/03-Best Practice/selecting-with-text.md:
```diff
@@ -0,0 +1,173 @@
+- **{2}**: Repeat the previous expression ([0-9]) two times.
+
+As the result, it will try to match every text that has a sequence starting with the digit 3 or 4, and then has any two digits in a row.
+
```
Remove double newline.

JohannesDienst-askui commented on docs/docs/general/05-Tooling/debug-with-annotation.md:
```diff
@@ -0,0 +1,118 @@
+
+- `annotate()`
+- `annotateInteractively()`
+
```
Remove newline.

JohannesDienst-askui reviewed (commented):
We have to make sure that the new API docs are also pushed to the askui-inference repository.

JohannesDienst-askui commented on docs/docs/general/02-Getting Started/next-steps.md:
```diff
@@ -0,0 +1,36 @@
+- Read the [Debug with Annotation](../05-Tooling/debug-with-annotation.md) to learn how to use askuis' **Image Annotation Feature** in order to get the best automation result.
+
+**Best Practice**
+- Read the [Speed Up Execution](../03-Best%20Practice/speed_up_execution.md) to understand more about the inference's performance of our AI model.
```
If we reverse the order it would be more logical I think.

JohannesDienst-askui commented:
This is great!

Some minor nitpickings:
- Make sure we also have a PR for askui-inference when you change something in the API as the pages are generated from there
- Please fix the spelling mistakes found by `npm run spellcheck-docs` in `docs` folder:
```
docs/general/05-Tooling/debug-with-annotation.md
     105:4-105:25  warning  `annotateInteractively` is misspelt                                                                        annotateinteractively  retext-spell
    106:44-106:54  warning  `fullscreen` is misspelt                                                                                   fullscreen             retext-spell
```
